[Bin hui] IP

binbinhui opened the pull request:
Pull request Testing

Impala36 committed:
Level 0. Greet

Impala36 committed:
Level 0. Greet

Impala36 committed:
Level 2. Add, List

Impala36 committed:
Level 3. Mark as Done

Impala36 committed:
Level 4. ToDos, Events, Deadlines

Impala36 committed:
Level 5. Handle Errors
A-AbstractClasses
A-Packages

binbinhui commented:
update code

ZhengShijieNUS commented on src/main/java/AddTasks.java:
```diff
@@ -0,0 +1,75 @@
+
+public class AddTasks {
```
AddTask sounds like action.  Maybe use TaskAddition will be better?

SJZheng-Abel replied:
AddTask sounds like action. Maybe use TaskAddition will be better?

dgc5213 replied:
Shall we use nouns for classes? Perhaps, we could consider using  Class name called: "TaskAddon".

SJZheng-Abel commented on src/main/java/AddTasks.java:
```diff
@@ -0,0 +1,75 @@
+public class AddTasks {
+
+
+    private int i;
```
i is usually used as a temporary variable, such as in the for loop, etc. Thus here I think maybe giving a specific name will be better and it will improve the readability.

dgc5213 replied:
Shall we use the variable name which has a sufficient level of details?

dgc5213 commented on src/main/java/AddTasks.java:
```diff
@@ -0,0 +1,75 @@
+        showTasks.setCheck();
+    }
+
+    public void eventTask(String event){
```
Shall we use verbs for methods/functions? Maybe change it to "setEventTask".

dgc5213 commented on src/main/java/AddTasks.java:
```diff
@@ -0,0 +1,75 @@
+        showTasks.setCheck();
+    }
+
+    public void deadLineTask(String deadLine){
```
Shall we use verbs for methods/functions? Maybe change it to "setDeadlineTask".

dgc5213 commented on src/main/java/DeadlineTasks.java:
```diff
@@ -0,0 +1,19 @@
+
+    ShowTasks showTasks = new ShowTasks();
+
+    public DeadlineTasks(){
```
It looks like an empty method or object. Shall we remove it?

dgc5213 commented on src/main/java/DeadlineTasks.java:
```diff
@@ -0,0 +1,19 @@
+    }
+
+
+    public void deadLineTask(String deadLine){
```
Shall we use verbs for methods/functions? The purpose of this method is not so straightforward to me. I am not sure what is this "deadLineTask" handling.  Shall we consider to edit the name and use the name to explain?

dgc5213 commented on src/main/java/EventTasks.java:
```diff
@@ -0,0 +1,20 @@
+
+    ShowTasks showTasks = new ShowTasks();
+
+    public EventTasks(){
```
It looks like an empty method or object. Shall we remove it?

dgc5213 commented on src/main/java/EventTasks.java:
```diff
@@ -0,0 +1,20 @@
+
+    }
+
+    public void eventTask(String event){
```
Shall we use verbs for methods/functions? The purpose of this method is not so straightforward to me. I am not sure what is this "eventTask" handling. Shall we consider to edit the name and use the name to explain?

dgc5213 commented on src/main/java/MakeDoneTasks.java:
```diff
@@ -0,0 +1,18 @@
+
+    ShowTasks showTasks = new ShowTasks();
+
+    public MakeDoneTasks(){
```
It looks like an empty method or object. Shall we remove it?

dgc5213 commented on src/main/java/SwitchTasks.java:
```diff
@@ -0,0 +1,76 @@
+
+public class SwitchTasks {
```
Shall we use nouns for class name? Maybe change it to "TasksManager".

dgc5213 reviewed (commented):
Overall, the program can support multiple functions such as add,  make as done, task listing. It would be good to further improve naming so that it would be much easier for other people to understand this program.  The layout of code is very clear.

Impala36 committed:
A-CodingQuality
Revamped many methods to look more readable,
avoided deep nesting,
happy path prominent,
refactored methods,
and SLAP.

Impala36 committed:
Level 6. Delete

Impala36 committed:
Level 7. Save

Impala36 committed:
A-JavaDoc
Added JavaDoc comments and removed redundant Ui method redBorderlines().

Impala36 committed:
Merge remote-tracking branch 'origin/branch-Level-7' into A-JavaDoc

# Conflicts:
#	src/main/java/duke/parser/Parser.java
#	src/main/java/duke/task/Deadline.java
#	src/main/java/duke/task/Event.java
#	src/main/java/duke/task/TaskList.java
#	src/main/java/duke/task/Todo.java
#	src/main/java/duke/ui/Style.java
#	src/main/java/duke/ui/Ui.java

Impala36 committed:
Edited JavaDoc comments.

binbinhui commented:
Please Merge

Impala36 committed:
A-MoreOOP
Extracted all commands into classes.

Impala36 committed:
Level 8. Date and Time
Added date and time to deadline and events, renamed taskDescription to line to avoid confusion with description (actual description of task).

Impala36 committed:
Level 9. Find

Impala36 committed:
Level 9. Find

Impala36 committed:
A-JUnit
Added JUnit tests, imported Gradle.

Impala36 committed:
A-UserGuide

Impala36 committed:
Merge pull request #1 from Impala36/branch-Level-9

Branch level 9

Impala36 committed:
Merge pull request #2 from Impala36/branch-A-JUnit

Branch-A-JUnit

Impala36 committed:
A-UserGuide
Some formats were not working.

Impala36 committed:
Merge pull request #3 from Impala36/branch-A-UserGuide

A-UserGuide

Impala36 committed:
Update README.md

Impala36 committed:
Update README.md

Impala36 committed:
A-Gradle
A-CheckStyle
Restructured code to 0 gradle checkstyle violations

Impala36 committed:
Level-10
Changed all commands and UI to return Strings for GUI

Impala36 committed:
Changed isDone tick and cross to D and X due to encoding issues

Impala36 committed:
Merge remote-tracking branch 'origin/branch-Level-10'

Impala36 committed:
Added gradle.yml

Impala36 committed:
BCD-Extension
Added D-Notes
Replaced all /n to /r due to DialogBox ellipsis issue
Removed unused text color functions

Impala36 committed:
Merge remote-tracking branch 'origin/branch-Level-10'

Impala36 committed:
Update UserGuide.md

Impala36 committed:
Move variable declarations closer to usage

Scope of variables are large and may cause issues in future.

Variables are declared just before their usage to reduce their scope and smoothen the flow of thought for the reader.

Let's declare the variables in the same scope and just before usage.

Also add and update README.md, DeveloperGuide.md,  UserGuide.md, DoneCommandTest.java and NoteCommandTest.java.

Impala36 committed:
Change permission of gradlew

Impala36 committed:
Merge remote-tracking branch 'origin/branch-Level-10'

Impala36 committed:
Remove redundant text-ui-test files

Program shifted to GUI so text UI tests are no longer required.

Impala36 committed:
Change permission of gradlew

Impala36 committed:
Merge pull request #5 from Impala36/branch-Level-10

Branch level 10

Impala36 committed:
Restyle to a medieval Duke theme

Makes the program more interesting to use.

Also update the README.md, DeveloperGuide.md, UserGuide.md.

Impala36 committed:
Merge remote-tracking branch 'origin/branch-Level-10' into branch-Level-10

Impala36 committed:
Change /r to \r and improve NoteCommand

Carriage returns were not displayed properly, and user inputting note without text is properly caught.

Impala36 committed:
Merge pull request #6 from Impala36/branch-Level-10

Branch level 10

Impala36 committed:
Update README.md

Impala36 committed:
Update UserGuide.md

Commit:
A-HigherCodingQuality

Commit:
Commit no track projects

binbinhui committed:
Update gradle.yml

Change the committer name

binbinhui committed:
Update UserGuide

Update UserGuide to the new logic

binbinhui committed:
Add files via upload

binbinhui committed:
Delete DukeGUI.png

binbinhui committed:
Delete DeveloperGuide.md

binbinhui committed:
Delete NoteSequenceDiagram.puml

binbinhui committed:
Delete NoteSequenceDiagram.png

binbinhui committed:
Delete style.puml

binbinhui committed:
Create DeveloperGuide.md

Create DeveloperGuide.md

binbinhui committed:
Add files via upload

binbinhui committed:
Add files via upload

binbinhui committed:
Update DeveloperGuide.md

binbinhui committed:
Delete storage.png

binbinhui committed:
Add files via upload

binbinhui committed:
Update DeveloperGuide.md

binbinhui committed:
Update DeveloperGuide.md

binbinhui committed:
Update DeveloperGuide.md

binbinhui committed:
Update README.md

Update README.md

binbinhui committed:
Create AboutMe.md

Create AboutMe.md

binbinhui committed:
Update README.md

binbinhui committed:
Update AboutMe.md

binbinhui committed:
Update AboutMe.md

binbinhui committed:
Add files via upload

binbinhui committed:
Delete UI.png

binbinhui committed:
Add files via upload

binbinhui committed:
Update AboutMe.md

binbinhui committed:
Update AboutMe.md

Commit:
Improve code quality and fix some bugs.

binbinhui committed:
Update README.md

binbinhui committed:
Delete Style.java

binbinhui committed:
Delete src/main/java/duke/database directory

binbinhui committed:
Delete src/main/java/duke/task directory

binbinhui committed:
Delete DaPeasant.png

binbinhui committed:
Delete NoteCommandTest.java

binbinhui committed:
Delete DatabaseTest.java

binbinhui committed:
Delete EventCommandTest.java

binbinhui committed:
Delete ParserTest.java
